snuba/settings/__init__.py

```diff
@@ -278,6 +255,45 @@
 COUNTER_WINDOW_SIZE = timedelta(minutes=10)
 
 
+# Slicing Configuration
+
+# Mapping of storage key to slice count
+# This is only for sliced storages
+SLICED_STORAGES: Mapping[str, int] = {}
+
+# Mapping storage key to a mapping of logical partition
+# to slice id
+LOGICAL_PARTITION_MAPPING: Mapping[str, Mapping[int, int]] = {}
```

Comment:
I think a direct mapping of logical partitions to slice id will be very impractical. 
There are hundreds of logical partitions (if we are talking about the logical shards from the original design). You probably want to map ranges of logical partitions to slice ids.:

```
Mapping[str, Mapping[Tuple[int, int], int]]
```

Or better:
```
PartitionRange = Tuple[int,int]
Mapping[str, Mapping[PartitionRange, int]]
```

Reply:
I see the merit of the approach you're proposing but I want to push back on doing this right now.

1. We already have code that depends on this `LOGICAL_PARTITION_MAPPING` (it may look new in this PR but that's because Riya moved it to consolidate with other slicing settings) and
2. We don't really have slicing in place yet, and in my mind the risk for this project is not "can we represent configuration in a friendly way?" but "how can we get slicing working at all?". Making this change now takes us further from addressing the risk, not closer.
3. This configuration is functionally equivalent to what you're proposing

I just don't see a huge difference in operational burden between:

```python
LOGICAL_PARTITION_MAPPING = { 
  "generic-metrics-distributions": {
     **{ x: 0 for x in range(0,128) },
     **{ x: 1 for x in range(128,256) }
   }
}
```

and

```python
LOGICAL_PARTITION_MAPPING = { "generic-metrics-distributions": {(0, 128): 0, (128,256): 1}}
```

but to make your change now requires taking focus off functional slicing changes to address this (IMO) aesthetic concern. I'm happy to ticket doing this later though

Reply:
Yeah, I've added a ticket to the epic (SNS-1766) to revisit this once we are more functionality-complete